solution/index.js

```diff
@@ -0,0 +1,316 @@
+}
+
+document.addEventListener("click" , textEditable);
+// make the text list editable
+function textEditable(event){
+    if(event.target.classList.contains("task"))
+    {
+        event.target.setAttribute("contenteditable" ,"true");
+        findEditList(event);
+    }   
+}
+
+let indexOfOldTask; // for the splice
+// find the correct iist of the edit text
+function findEditList(event){
+    const li = event.target;
+    const buttonId = li.parentElement.parentElement.parentElement.querySelector("button").id;
+    if(buttonId === "submit-add-to-do")
+    {
+        indexOfOldTask = toDoStorage.indexOf(li.innerText); // for the splice
+        li.addEventListener("blur" , findEditedStorageList);    
+    }
+    else if(buttonId === "submit-add-in-progress")
+    {
+        indexOfOldTask = inProgressStorage.indexOf(li.innerText); // for the splice
+        li.addEventListener("blur" , findEditedStorageList);    
+    }
+    else if(buttonId === "submit-add-done")
+    {
+        indexOfOldTask = doneStorage.indexOf(li.innerText); // for the splice
+        li.addEventListener("blur" , findEditedStorageList);    
+    }
+}
+// send the correct arr to the saveEditText function
+function findEditedStorageList(event){
+    const li = event.target;
+    const buttonId = li.parentElement.parentElement.parentElement.querySelector("button").id;
+    if(buttonId === "submit-add-to-do")
+    {
+        saveEditText(li , toDoStorage);
+    }
+    if(buttonId === "submit-add-in-progress")
+    {
+        saveEditText(li , inProgressStorage);
+    }
+    if(buttonId === "submit-add-done")
+    {
+        saveEditText(li , doneStorage);
+    }
+}
+//save the edited text in the local storage
+function saveEditText (input  , storageList){
+    if(input.innerText === "") return alert("You must enter content");
+    storageList.splice(indexOfOldTask,1,input.innerText);
+    localStorage.setItem("tasks" ,JSON.stringify(localStorageObj));
+}
+// A function that takes an event, what you want to apply the event to, and what will happen when the event happens
+function addGlobalEventListener( type , selector , callback){
+    document.addEventListener(type , e => {
+    if (e.target.matches(selector)) callback(e);
+    })
+}
+
+let elementThatMove; // the element that we need to remove
+// when the mouse over <li> addEventKeydown function will play
+addGlobalEventListener("mouseover" , "li" , addEventKeydown);
+// start to listen keydown event
+function addEventKeydown(element){
+   elementThatMove = element.target;
+   document.addEventListener("keydown" , findKeyList);
+}
+//find the ul we want to move the task to
+function findKeyList(event){
+    if(event.altKey && event.key === "1" )
+    {
+        moveTask(toDoListEl , toDoStorage);
+    }
+    else if(event.altKey && event.key === "2" )
+    {
+        moveTask(inProgressListEl , inProgressStorage);
+    }
+    else if(event.altKey && event.key === "3" )
+    {
+        moveTask(doneListEl , doneStorage);
```

Comment:
It seems the only thing that changes between these `if` blocks is the parameters to `moveTask`.
Why not replace the entire conditional bonanza with an object that maps the key number to the respective parameters?